### ISSUE TITLE
FloatVector constuctor behaves strangely when given np.int array.

### ISSUE THREAD

**Issue by komarkdev** [rpy2/rpy2#702]
Hi! 

We do the following:
```
from rpy2 import robjects
robjects.FloatVector(np.arange(1,10))
```
And get:

    Out[3]: 
    R object with classes: ('numeric',) mapped to:
    [0.000000, 0.000000, 0.000000, 0.000000, ..., 0.000000, 0.000000, 0.000000, 0.000000]

which is not what one would expect. 

Just to be sure:
```
np.array(robjects.FloatVector(np.arange(1,10)))  
```
    Out[4]: 
    array([4.9e-324, 9.9e-324, 1.5e-323, 2.0e-323, 2.5e-323, 3.0e-323,
           3.5e-323, 4.0e-323, 4.4e-323])

This of course works ok with explicit casting:
```
robjects.FloatVector(np.arange(1,10).astype(np.float))
```
    Out[5]: 
    R object with classes: ('numeric',) mapped to:
    [1.000000, 2.000000, 3.000000, 4.000000, ..., 6.000000, 7.000000, 8.000000, 9.000000]

But the current behavior seems quite problematic.  

The strangest thing is that this code used to work properly (i.e. doing the correct casting ) about 6 month ago on the same system. The only change I can think of is perhaps R itself got updated somehow. 

Ubuntu 18.04, Python 3.6, rpy2 3.3.3 (but older versions have this issue too), R 3.4.4

**Comment by lgautier** [comment]
Thanks. This is not the what should happen.

The issue seems to be caused by `int64` value (that R cannot handle) as the following appears to be working as expected:

```python
robjects.FloatVector(np.arange(1,10, dtype='int32'))
```

**Comment by lgautier** [comment]
rpy2 uses Python's buffer interface to speed up the process of getting data in `numpy` arrays into R arrays, but something unexpected seems to be happening with numpy:

```python
>>> a = np.arange(1,10, dtype='int64')
>>> a                                                                                                  
array([1, 2, 3, 4, 5, 6, 7, 8, 9])
>>> np.frombuffer(a, dtype='d')
array([4.9e-324, 9.9e-324, 1.5e-323, 2.0e-323, 2.5e-323, 3.0e-323,
       3.5e-323, 4.0e-323, 4.4e-323])
``` 

With `int32` the following is happening:

```python
>>> a = np.arange(1,10, dtype='int32')                                                                 
>>> np.frombuffer(a, dtype='d')                                                                        
ValueError                                Traceback (most recent call last)
<ipython-input-21-be078581644c> in <module>
1 np.frombuffer(a, dtype='d')
ValueError: buffer size must be a multiple of element size
```

This is the expected behavior: the memoryview trick cannot be used and rpy2 falls back to a slower data copy mechanism.

**Comment by lgautier** [comment]
There is in fact a check for consistency of itemsize:
https://github.com/rpy2/rpy2/blob/master/rpy2/rinterface_lib/sexp.py#L557

...but `sizeof(int64)` == `sizeof(double)` on a 64bit system. :/

**Comment by lgautier** [comment]
Since checking that itemsize are the same is not enough, adding a compatibility check here is likely the way to fix this:
https://github.com/rpy2/rpy2/blob/master/rpy2/rinterface_lib/sexp.py#L587

**Comment by lgautier** [comment]
This is closed in `master` and in branch `v3.3.x`. The fix will be available with rpy2-3.3.4.
